### PR TITLE
improve quarantine view

### PR DESCRIPTION
**Pull request by ghost**
- Handle user extensions when filtering quarantine (#55)
  To support this feature two new settings have been added to Modoboa\Parameters\Amavis localpart_is_case_sensative and recipient_delimiter the value of these settings must match what the administrator has set in their amavisd.conf.

  When a recipient_delimiter is specified on the settings page quarantined e-mails sent to user extensions will appear in the users quarantine page, this also works for any aliases associated with the user.

- make quarantine search case insensitive

### CODE REVIEW

File: modoboa_amavis/lib.py [hunk 1]
```diff
@@ -395,3 +397,80 @@ def setup_manual_learning_for_mbox(mbox):
...
+
+def split_address(address):
+    assert isinstance(address, six.text_type),\
+        "address should be of type %s" % six.text_type.__name__
+    if "@" not in address:
+        local_part = address
+        domain = ""
+    else:
+        local_part, domain = address.split("@", 1)
+    return (local_part, domain)
```
Comment (tonioo) [inline modoboa_amavis/lib.py]: Duplicate of https://github.com/modoboa/modoboa/blob/master/modoboa/lib/email_utils.py#L265 ?

File: modoboa_amavis/lib.py [hunk 2]
```diff
@@ -395,3 +397,80 @@ def setup_manual_learning_for_mbox(mbox):
...
+        pass
+    elif (
+        delimiter
+        and delimiter in local_part
+        and local_part[0] != delimiter
+        and local_part[-1] != delimiter
+    ):
+        local_part, extension = local_part.split(delimiter, 1)
+
+    return (local_part, extension)
```
Comment (tonioo) [inline modoboa_amavis/lib.py]: Same here. Maybe we could improve the function available in modoboa.lib ?

**Review by tonioo — COMMENTED** [review]
It's a great day ! :-)

File: modoboa_amavis/lib.py (hunk 1 above)
Comment (fyfe) [reply modoboa_amavis/lib.py]: fixed, see modoboa/modoboa#1360

File: modoboa_amavis/lib.py (hunk 2 above)
Comment (fyfe) [reply modoboa_amavis/lib.py]: fixed, see modoboa/modoboa#1360

**Comment by fyfe** [comment]
Depends on modoboa/modoboa#1360

**Comment by fyfe** [comment]
this is now ready to be merged

**Pull request merged**